### ISSUE TITLE
probe: pdf injection

### ISSUE THREAD

**Issue by leondz** [leondz/garak#841]
https://kai-greshake.de/posts/inject-my-pdf/

**Comment by leondz** [comment]
waiting on #1254

**Comment by leondz** [comment]
#1254 merged! no longer waiting

**Comment by oujezdsky** [comment]
Hey, I'd like to pick this up. I plan to write a probe based on documented vulnerabilities on https://kai-greshake.de/posts/inject-my-pdf/.
I'm a little unsure about the best approach for the generators, should I implement PDF support directly into for example HuggingFace generator (subclassing the `Model`)? or do you have a different pattern in mind for handling PDF file inputs?"

**Comment by francose** [comment]
Hey — if it's still available I'd like to pick this up. I noticed @oujezdsky expressed interest back in December but there hasn't been a PR since.

I've already done some local validation:

- Built a POC that generates injected PDFs using four techniques from the original research (white-on-white text, micro font, metadata fields, overlapping text layers)
- Confirmed that 3/4 techniques are extractable by PyMuPDF (the kind of parser LLM pipelines typically use), while remaining invisible to basic tools like `pdftotext`
- Reviewed the `Message(data_path=..., data_type=("application","pdf"))` pattern from #1254 and the existing `visual_jailbreak.py` / `latentinjection.py` probes for structural reference

My planned approach:
1. Probe generates injected PDFs at runtime (reportlab) with configurable techniques and payloads
2. Uses `TriggerListDetector` to check if the model followed the hidden instructions
3. Includes a clean control PDF for baseline comparison
4. Tags: `owasp:llm01`, `avid-effect:security:S0403`

Happy to adjust the approach based on maintainer feedback.
